packages/debugger/src/panels/sources/model.ts

```diff
@@ -16,13 +20,47 @@ export class SourcesModel implements IDebugger.Model.ISources {
    * @param options The Sources.Model instantiation options.
    */
   constructor(options: SourcesModel.IOptions) {
-    this.currentFrameChanged = options.currentFrameChanged;
+    this._currentSource = null;
+    this._currentFrame = null;
+
     this._displayRegistry =
       options.displayRegistry ?? new DebuggerDisplayRegistry();
+
+    if (options.editorServices?.mimeTypeService) {
+      this._mimeTypeService = options.editorServices?.mimeTypeService;
+    }
+
+    options.currentFrameChanged.connect(async (_, frame) => {
+      if (frame) {
+        const displayPath = this.getDisplayName(frame);
+
+        const source = await options.getSource({
+          sourceReference: 0,
+          path: frame?.source?.path
+        });
+
+        if (source) {
+          const { content, mimeType } = source;
+          const editorMimeType =
+            mimeType ||
+            this._mimeTypeService.getMimeTypeByFilePath(
+              frame.source?.path ?? ''
+            );
+
+          this._currentFrame = frame;
+
+          this.currentSource = {
+            content: content,
+            mimeType: editorMimeType,
+            path: displayPath
+          };
+        }
+      }
+    });
   }
 
   /**
-   * Signal emitted when the current frame changes.
+   * Signal emitted when the current frame changes. @deprecated since 4.6.0, will be removed in 5.0
```

Comment:
For a better visibility

```suggestion
   * Signal emitted when the current frame changes.
   * 
   * @deprecated since 4.6.0, will be removed in 5.0.
```

And the same comment should be in the interface too, `IDebugger.Model.ISources`

Reply:
The deprecation docstring has not yet been added to the interface.